From 456257d346b295982761d33f4e24e07a9feee74d Mon Sep 17 00:00:00 2001
From: Marco Felsch <mfe@pengutronix.de>
Date: Sat, 26 Nov 2022 08:00:39 +0100
Subject: [PATCH 1/2] build(makefile): add helper to detect linker options

This is a small helper to check for possible linker options. If the
linker supports the requested option it is returned and if not nothing
will be returned, e.g.:

  TF_LDFLAGS += $(call ld_option, --no-warn-rwx-segments)

can be called unconditional.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>
---
 make_helpers/build_macros.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/make_helpers/build_macros.mk b/make_helpers/build_macros.mk
index b89d87ea6087..9f4f70dbe42f 100644
--- a/make_helpers/build_macros.mk
+++ b/make_helpers/build_macros.mk
@@ -67,6 +67,12 @@ $(foreach d,$(0-9),$(eval __numeric := $(subst $(d),,$(__numeric))))
 $(if $(__numeric),$(error $(1) must be numeric))
 endef
 
+# Convenience function to check for a given linker option. An call to
+# $(call ld_option, --no-XYZ) will return --no-XYZ if supported by the linker
+define ld_option
+	$(shell if $(LD) $(1) -v >/dev/null 2>&1; then echo $(1); fi )
+endef
+
 # IMG_LINKERFILE defines the linker script corresponding to a BL stage
 #   $(1) = BL stage (2, 30, 31, 32, 33)
 define IMG_LINKERFILE
-- 
2.30.2

